### PR TITLE
GHCR image push- tfc-bacckup-b2

### PR DESCRIPTION
**Pull request by Praveenraj-K**
-Pushing the images in both Docker Hub & GitHub Container Registry "GHCR.io"
-Only initiate when there is a commit in the main branch using the workflow trigger.
-Removed the latest tag and added the version based tagging

### CODE REVIEW

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -17,19 +19,31 @@ jobs:
...
           tags: |
-            type=ref,event=branch            
-            type=semver,pattern={{version}}
-            type=semver,pattern={{major}}.{{minor}}
-            # set 'latest' tag for main branch
-            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=ref,event=branch  # Tag with branch name
+            type=semver,pattern={{version}},enable=${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
+            type=semver,pattern={{major}}.{{minor}},enable=${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
+            type=semver,pattern={{major}},enable=${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
```
Comment (briskt) [inline .github/workflows/build-and-publish.yml]: I think the `enable` may not be necessary. The [docs](https://github.com/docker/metadata-action?tab=readme-ov-file#typesemver) indicate that a semver only applies to a `push` event with a valid semver tag.
```suggestion
            type=semver,pattern={{version}}
            type=semver,pattern={{major}}.{{minor}}
            type=semver,pattern={{major}}
```

File: .github/workflows/build-and-publish.yml [hunk 2]
```diff
@@ -17,19 +19,31 @@ jobs:
...
+          labels: |
+            org.opencontainers.image.source=${{ github.repository }}
+            org.opencontainers.image.revision=${{ github.sha }}
+            org.opencontainers.image.created=${{ github.event.head_commit.timestamp }}
```
Comment (briskt) [inline .github/workflows/build-and-publish.yml]: I'm not certain, but I think these labels may be added by default. Did you try it without adding them explicitly?
```suggestion
```

**Review by briskt — APPROVED** [review]
This looks good! I added some suggestions for simplification.

File: .github/workflows/build-and-publish.yml (hunk 2 above)
Comment (Praveenraj-K) [reply .github/workflows/build-and-publish.yml]: i'll this in the develop branch and keep you posted.

File: .github/workflows/build-and-publish.yml [hunk 3]
```diff
@@ -2,10 +2,12 @@ name: Build and Publish
 
 on:
   push:
+    branches: [ 'main' ]  # Trigger on push to 'main'
+    tags: [ '[0-9]+.[0-9]+.[0-9]' ]  # Trigger on version tags like '0.1.0'
```
Comment (briskt) [inline .github/workflows/build-and-publish.yml]: Without this final `+`, a version like 1.2.12 wouldn't pass.
```suggestion
    tags: [ '[0-9]+.[0-9]+.[0-9]+' ]  # Trigger on version tags like '0.1.0'
```

**Review by briskt — APPROVED** [review]

**Pull request merged**